examples/AspNetCore/Controllers/WeatherForecastController.cs

```diff
@@ -54,11 +60,23 @@ public IEnumerable<WeatherForecast> Get()
         })
         .ToArray();
 
+        // Log using ILogger API.
         this.logger.LogInformation(
             "WeatherForecasts generated {count}: {forecasts}",
             forecast.Length,
             forecast);
 
+        // Log using LogEmitter API.
+        var logRecord = LogRecordPool.Rent();
+
+        logRecord.CategoryName = "WeatherForecasts";
+        logRecord.LogLevel = LogLevel.Information;
+        logRecord.FormattedMessage = "WeatherForecasts generated.";
+        logRecord.StateValues = new List<KeyValuePair<string, object?>>() { new KeyValuePair<string, object?>("count", forecast.Length) };
+        logRecord.SetActivityContext(Activity.Current);
+
+        this.logEmitter.Log(logRecord);
```

Comment:
What would be the semantic / expected behavior for the following code snippet? Or it is considered as a user bug (they need to clearly understand the ownership contract)?

```csharp
var logRecord = LogRecordPool.Rent();
...
logRecord.FormattedMessage = "old";
this.logEmitter.Log(logRecord);
logRecord.FormattedMessage = "new";
this.logEmitter.Log(logRecord);
```

Reply:
I'm bit concerned about the API design here. For example, the timestamp was generated during `Rent()`?

Reply:
Pushed a new iteration. Pooling is now encapsulated inside `OpenTelemetryLogger` and `LogEmitter` so the above is not possible. Also added a way to pass up to 8 tags/attributes without allocation. Check it out, LMK your thoughts on the new version.